stress/src/geneva_exporter.rs

```diff
@@ -290,7 +295,33 @@ async fn async_main(
             ThroughputTest::run_continuous("Geneva Upload", config, move || {
                 let client = client.clone();
                 let logs = logs.clone();
-                async move { client.upload_logs(&logs).await }
+                async move {
+                    let batches = client
+                        .encode_and_compress_logs(&logs)
+                        .map_err(std::io::Error::other)?;
+
+                    // All batch uploads happen within the same async task that called `encode_and_compress_logs`
+                    // Note: The batches are uploaded concurrently 10 at a time, however the test setup currently doesn't have
+                    // multiple concurrent operations.
+                    use futures::stream::{self, StreamExt};
+                    let errors: Vec<String> = stream::iter(batches)
```

Comment:
This seems odd. We now have a stream of futures which further get used to produce another stream of futures in the `run_continuous` method. I think we need to fix the test.